File: Java/comments/src/main/java/ua/edu/chnu/comments/services/CommentService.java
package ua.edu.chnu.comments.services;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import ua.edu.chnu.comments.exceptions.CommentNotFoundByIdException;
import ua.edu.chnu.comments.models.Comment;
import ua.edu.chnu.comments.repositories.CommentRepository;

import java.util.List;

@RequiredArgsConstructor
@Service
public class CommentService {
    private final CommentRepository repository;

    public Comment create(Comment created) {
        return repository.save(created);
    }

    public List<Comment> readAll() {
        return repository.findAll();
    }

    public Comment read(int id) {
        return repository.findById(id).orElseThrow(() -> new CommentNotFoundByIdException(id));
    }

    public void update(int id, Comment comment) {
        Comment updated = read(id);

        updated.setAuthor(comment.getAuthor());
        updated.setContent(comment.getContent());

        repository.save(updated);
    }

    public void delete(int id) {
        repository.delete(read(id));
    }
}

File: api/Comments/BLL/Services/CommentService.cs
using BLL.Exceptions;
using DAL.Data;
using DAL.Models;

namespace BLL.Services;

public class CommentService(CommentRepository repository)
{
    public async Task<int> CreateAsync(Comment comment)
    {
        int id = await repository.CreateAsync(comment);
        return id;
    }

    public async Task<IEnumerable<Comment>> ReadAllAsync()
    {
        var comments = await repository.ReadAllAsync();
        return comments;
    }

    public async Task<Comment> ReadAsync(int id)
    {
        Comment? comment = await repository.ReadAsync(id);
        return comment ?? throw new CommentNotFoundException("comment not found by id: " + id);
    }

    public async Task UpdateAsync(int id, Comment comment)
    {
        await ReadAsync(id);

        await repository.UpdateAsync(id, comment);
    }

    public async Task DeleteAsync(int id)
    {
        await ReadAsync(id);

        await repository.DeleteAsync(id);
    }
}